weave/trace/op.py

```diff
@@ -62,7 +59,7 @@ def _watched_call(self, *args: Any, **kwargs: Any) -> Any:
         )
         try:
             with run_context.current_run(run):
-                res = self.resolve_fn(**inputs)
+                res = self.resolve_fn(*args, **kwargs)
```

Comment:
I think this may be a problem that somehow the tests aren't catching.

we call client.save_nested_objects on inputs_with_defaults. That method attaches refs to certain objects that may be contained in inputs.

Are args/kwargs guaranteed to have references to the same values that inputs/inputs_with_defaults has?

Thinking about it, maybe its fine, but I'm not 100% confident.

Reply:
I thought about this as well. I am pretty sure the whole chain here is pass by reference:
```
inputs = self.signature.bind(*args, **kwargs).arguments
inputs_with_defaults = _apply_fn_defaults_to_inputs(self.resolve_fn, inputs)
client.save_nested_objects(inputs_with_defaults)
res = self.resolve_fn(*args, **kwargs) # here, the values in these containers should be the same values passed to `save_nested_objects`
```